torch/onnx/_internal/exporter/_compat.py

```diff
@@ -225,6 +316,17 @@ def export_compat(
                 )
             if f is None:
                 raise TypeError("f must be provided when fallback is enabled") from e
+            if dynamic_shapes is not None and dynamic_axes is None:
+                try:
+                    dynamic_axes = _from_dynamic_shapes_to_dynamic_axes(
+                        dynamic_shapes=dynamic_shapes,
+                        input_names=input_names,
+                    )
+                except Exception as e:
+                    raise RuntimeError(
+                        "# Failed to convert 'dynamic_shapes' to 'dynamic_axes'. "
+                        "Please provide 'dynamic_axes' directly for fallback. "
```

Comment:
This can be confusing because if  the user just got a recommendation for providing dynamic_shapes, it may be confusing and frustrating to get asked for dynamic axes again. Is this required? If so, I would provide a more detailed explanation on why this is needed, and point out that they should still keep the dynamic shapes argument. Also, If the fallback is triggered due to incorrect dynamic shapes argument, the right thing to do would be fixing the dynamic shapes. I wonder how we can communicate that to users as well.